internal/version/version.go

```diff
@@ -17,7 +17,7 @@ package version
 
 var (
 	// Version is the current version of the oras.
-	Version = "1.1.0-rc.1"
+	Version = "1.1.0-rc1+dev"
 	// BuildMetadata is the extra build time data
 	BuildMetadata = "unreleased"
```

Comment:
@sajayantony  We already have extra build time data here, the full version is `<Version>+<BuildMetadata>`
e.g. 
- if you build from this commit, the full version string will be `1.1.0-rc1+dev+unreleased`. 
- Without this code change in this PR, the full version string will be `1.1.0-rc.1+unreleased`, the latter already implies the former with less duplicated info.

Reply:
Definition quotes from [SemVer 2.0.0](https://semver.org/#semantic-versioning-specification-semver):

> 10. Build metadata MAY be denoted by appending a plus sign and a series of dot separated identifiers immediately following the patch or pre-release version. Identifiers MUST comprise only ASCII alphanumerics and hyphens [0-9A-Za-z-]. Identifiers MUST NOT be empty. Build metadata MUST be ignored when determining version precedence. Thus two versions that differ only in the build metadata, have the same precedence. Examples: 1.0.0-alpha+001, 1.0.0+20130313144700, 1.0.0-beta+exp.sha.5114f85, 1.0.0+21AF26D3----117B344092BD.

Reply:
@shizhMSFT should we up the version then?

Reply:
The version is already updated in https://github.com/oras-project/oras/commit/42650e5778283be7e4926f123f183db12dfbacaa before the branch cut. @sajayantony Are you suggesting we should do the version update for the next version? We don't have a procedure for that and might be worth doing after releasing.

Reply:
Let's reset the releasing pipeline of `v1.1.0-rc.1`.

Reply:
See details in #1013